src/migration/1726584940063-SourceCategoryPriority.ts

```diff
@@ -0,0 +1,66 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class SourceCategoryPriority1726584940063 implements MigrationInterface {
+  name = 'SourceCategoryPriority1726584940063';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `DELETE FROM "source_category" WHERE "title" IN('Data', 'Cloud', 'DevOps', 'General')`,
+    );
+    await queryRunner.query(`
+      INSERT INTO "source_category"
+      (title, enabled)
+      VALUES
+      ('Basics', true),
+      ('AI', true),
+      ('DevRel', true),
+      ('Open Source', true),
+      ('DevOps & Cloud', true)
+      ON CONFLICT DO NOTHING;
+    `);
+    await queryRunner.query(
+      `ALTER TABLE "source_category" ADD "priority" integer`,
+    );
+    await queryRunner.query(
+      `
+        DO $$
+        DECLARE
+            categories TEXT[] := ARRAY['Basics','Web','Mobile','DevOps & Cloud','AI','Games','DevTools','Career','Open Source','DevRel','Fun'];
+            i INT;
+        BEGIN
+            -- Iterate over the array and update the table
+            FOR i IN 1..array_length(categories, 1) LOOP
+                UPDATE source_category
+                SET priority = i
+                WHERE slug = trim(BOTH '-' FROM regexp_replace(lower(trim(COALESCE(LEFT(categories[i],100),''))), '[^a-z0-9-]+', '-', 'gi'));
+            END LOOP;
+        END $$;
+      `,
+    );
+  }
```

Comment:
Just thinking can we do this inserts as a once off, if we ever add new categories it will mean we need to again fight with migrations. Non blocking, up to you whatever you think its best.

Reply:
Right. Yeah, really hard to decide if want to drop everything here, then just prepare an insert script along with the ordering one.

Reply:
That might actually be the better idea as I can export what will be created in prod then make those data as the seeds to make it consistent if we want to do more seedings that involve this table. Rather than generating new ids every time we run migrations locally.